### PR TITLE
Update ZFS collector for Linux

### PR DESCRIPTION
**Pull request by richardelling**
Updated Linux ZFS agent to be compatible with ZFSonLinux v0.8.0-rc1
+ dbufcachestats file renamed to dbufstats
+ added pool health, compatible with FreeBSD's tags
+ added optional objset (dataset) stats
+ more comprehensive tests for Linux-specific stats

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

Signed-off-by: Richard Elling <Richard.Elling@RichardElling.com>

### CODE REVIEW

File: plugins/inputs/zfs/zfs_linux.go [hunk 1]
```diff
@@ -60,30 +66,122 @@ func gatherPoolStats(pool poolInfo, acc telegraf.Accumulator) error {
...
 	if keyCount != len(values) {
-		return fmt.Errorf("Key and value count don't match Keys:%v Values:%v", keys, values)
+		return fmt.Errorf("key and value count don't match Keys:%v Values:%v", keys, values)
 	}
 
 	tag := map[string]string{"pool": pool.name}
 	fields := make(map[string]interface{})
 	for i := 0; i < keyCount; i++ {
-		value, err := strconv.ParseInt(values[i], 10, 64)
+		value, err := strconv.ParseUint(values[i], 10, 64)
```
Comment (glinton) [inline plugins/inputs/zfs/zfs_linux.go]: Will the uint conversion cause issues for older versions of influx?

Comment (richardelling) [reply plugins/inputs/zfs/zfs_linux.go]: good question! In my testing telegraf is emitting `field=###i`, regardless of the setting of `influx_uint_support` in `telegraf.conf`

Comment (richardelling) [reply plugins/inputs/zfs/zfs_linux.go]: I did more testing and it does properly handle unsigned/signed based on `influx_uint_support`, but for some reason, not when running in `--test` mode... I'll leave that for another day

**Comment by richardelling** [comment]
I'll rebase this weekend

**Comment by tomtastic** [comment]
Has this gone dead?

**Comment by danielnelson** [comment]
Closing due to inactivity.

**Pull request closed without merging**